controller/server/gui.ml

```diff
@@ -263,57 +261,125 @@ module NetworkGui = struct
       >|= List.filter (fun s -> not internet_connected || s |> Service.is_connected)
     in
 
+    let blur_service_proxy_password s =
+      let open Service in
+      let blur p = Proxy.validate p |> Option.map (Proxy.to_string ~hide_password:true) in
+      { s with proxy = s.proxy |> Base.Fn.flip Option.bind blur }
+    in
+
+    let after_restart_proxy =
+      services
+      |> List.find_map (fun s ->
+          if s.Service.autoconnect then
+            Option.bind s.Service.proxy Proxy.validate
+          else
+            None)
+    in
```

Comment:
I tried to use [connman-proxy](https://github.com/dividat/playos/pull/54/files#diff-a00760ca559b79269f2bd1c6d1562867820c1e351358198ec6eb150d2d51f043) there, but I get a Unix pipe error. I was not able to get a more precise error, I wonder if it’s because of a dbus conflict.

Reply:
Not relevant anymore.